diff --git a/gaudim/webrtc-client/src/components/DiagnosticsComponent/DiagnosticsView.js b/gaudim/webrtc-client/src/components/DiagnosticsComponent/DiagnosticsView.js
index bf1675fed0..49dc8bf4ad 100644
--- a/gaudim/webrtc-client/src/components/DiagnosticsComponent/DiagnosticsView.js
+++ b/gaudim/webrtc-client/src/components/DiagnosticsComponent/DiagnosticsView.js
@@ -1,4 +1,4 @@
-import React, { useEffect } from "react";
+import React, { useEffect, useRef } from "react";
 import Dialog from "@material-ui/core/Dialog";
 import data from "../../phone.json";
 import DialogTitle from "@material-ui/core/DialogTitle";
@@ -24,7 +24,9 @@ import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
 import CheckCircleIcon from "@mui/icons-material/CheckCircle";
 import ErrorIcon from "@mui/icons-material/Error";
 import CloseIcon from "@mui/icons-material/Close";
-
+import { createRef, useState } from 'react'
+import html2canvas from "html2canvas";
+import { useScreenshot } from 'use-react-screenshot'
 
   /*
   const [webSocketStatus, setWebSocketState] = React.useState(true);
@@ -80,6 +82,22 @@ import CloseIcon from "@mui/icons-material/Close";
   var relexiveConnectivityDescription = "relexiveConnectivityDescription";
   var browserVersion = 'BrowserVersion';
 
+  var networkImageUrl;
+
+
+function downLoadNetworkImage() {
+  const a = document.createElement("a");
+  a.href = networkImageUrl;
+  a.download = "DiagnosticsView.png";
+  document.body.appendChild(a);
+  // wait for the link to be added to the document
+  window.requestAnimationFrame(function () {
+    var event = new MouseEvent('click');
+    a.dispatchEvent(event);
+    document.body.removeChild(a);
+  });  
+
+};
 
 function DiagnosticsView({
   diagnosticValues,
@@ -91,6 +109,37 @@ function DiagnosticsView({
   const [open, setOpen] = React.useState(true);
   //const [browserVersion, setBrowserVersion] = React.useState('chrome');
   const channel = new BroadcastChannel("app-data");
+  const useReactRef = createRef(null)
+  const [image, takeScreenshot] = useScreenshot({
+    type: "image/jpeg",
+    quality: 1.0,
+  });
+
+  const download = (image, { name = "DiagnosticViews", extension = "jpg" } = {}) => {
+
+    /*var iakImg = document.getElementById('img')
+    image.src=iakImg.src + '?' + new Date().getTime();
+    imgObj.setAttribute('crossOrigin', 'anonymous');
+    image.onload = function () {
+      ctx.drawImage(image,0,0)
+      var data = ctx.getImageData(0,0,600,400)
+    } */   
+    const a = document.createElement("a");
+    a.href = image;
+    //a.download = createFileName(extension, name);
+    a.download = name + "." + extension;
+    document.body.appendChild(a);
+    // wait for the link to be added to the document
+    window.requestAnimationFrame(function () {
+      var event = new MouseEvent('click');
+      a.dispatchEvent(event);
+      document.body.removeChild(a);
+    }); 
+  };
+
+  const downloadScreenshot = () => takeScreenshot(useReactRef.current).then(download);
+
+
 
   const handleDiagnosticOps = (e) => {
     /**
@@ -141,13 +190,28 @@ function DiagnosticsView({
     userRegistrationStatus = false;
     userRegistrationDescription = "error in user registration check"
   }
+
+  var myCanvas;
+  const connectionsContainer = useRef(null);
+  const rootContainer = document.getElementById("root");
+  useEffect(() => async (dispatch) =>  {
+    /*console.log("devicesContainer..", connectionsContainer.current);
+    console.log("rootContainer..", rootContainer);
+    await html2canvas(rootContainer).then(myCanvas => {
+      document.body.appendChild(myCanvas);
+      networkImageUrl = myCanvas.toDataURL("image/png", 1.0);
+      console.log("networkImageUrl = ", networkImageUrl)      
+    });*/
+  },[])  
+
   return (
-    <div className="container">
+    <div  ref={useReactRef} className="container">
       <Dialog
         open={open}
         onClose={handleClose}
         aria-labelledby="alert-dialog-title"
         aria-describedby="alert-dialog-description"
+        ref={connectionsContainer} 
       >
         <DialogTitle id="alert-dialog-title">
           <div className="flex-container-top">
@@ -363,7 +427,11 @@ function DiagnosticsView({
             </CardContent>
           </Card>
         </DialogContent>
-        <p></p>
+        <span>
+          <div style='display: none'>
+            <img id='img' src='DiagnsticsView.jpg' width='600' height='400' />
+          </div>
+        </span>
         <Card sx={{ maxWidth: 350 }}>
         <div className="flex-container">
           &nbsp;           
@@ -393,6 +461,16 @@ function DiagnosticsView({
             onClick={handleDiagnosticsOpsSaveLogs}
           />
           </div>
+
+          <div className="flex-child">
+          <input
+            type="submit"
+            size="small"
+            value="Download UseReact"
+            onClick={downloadScreenshot}
+          />
+        </div>
+
           </div>
           <br/>
           <br/>
