diff --git a/gaudim/webrtc-client/src/components/DiagnosticsComponent/DiagnosticsDevices.js b/gaudim/webrtc-client/src/components/DiagnosticsComponent/DiagnosticsDevices.js
index 6908d9abd5..18a0b12e2a 100644
--- a/gaudim/webrtc-client/src/components/DiagnosticsComponent/DiagnosticsDevices.js
+++ b/gaudim/webrtc-client/src/components/DiagnosticsComponent/DiagnosticsDevices.js
@@ -1,4 +1,4 @@
-import React, { useEffect } from "react";
+import React, { useEffect, useRef } from "react";
 import Dialog from "@material-ui/core/Dialog";
 import data from "../../phone.json";
 import DialogTitle from "@material-ui/core/DialogTitle";
@@ -26,11 +26,23 @@ import CheckCircleIcon from "@mui/icons-material/CheckCircle";
 import ErrorIcon from "@mui/icons-material/Error";
 import CloseIcon from "@mui/icons-material/Close";
 import { yellow } from "@material-ui/core/colors";
+import { useState } from 'react'
+import html2canvas from "html2canvas";
+import { createRef } from 'react'
+import { useScreenshot } from 'use-react-screenshot'
 
+//import ViewShot from "react-native-view-shot";
+//import { captureRef } from "react-native-view-shot";
 
 var micStatus = true;
 var speakerStatus = true;
 var browserVersion = 'BrowserVersion';
+var deviceImageUrl;
+var myCanvas;
+
+
+
+
 
 function DiagnosticsDevices({
   diagnosticValues,
@@ -67,6 +79,91 @@ function DiagnosticsDevices({
     }
   };
 
+  const devicesContainer = useRef(null);
+  const rootContainer = document.getElementById("root");
+  const viewShotRef = useRef(null);
+  const useReactRef = createRef(null)
+  const [image, takeScreenshot] = useScreenshot({
+    type: "image/jpeg",
+
+    quality: 1.0
+  });
+
+  function downLoadImageUrl(imageUrl) {
+    console.log("downLoadImageUrl..", imageUrl);
+
+    const a = document.createElement("a");
+    a.href = imageUrl;
+    a.download = "DiagnosticDevices.png";
+    document.body.appendChild(a);
+    // wait for the link to be added to the document
+    window.requestAnimationFrame(function () {
+      var event = new MouseEvent('click');
+      a.dispatchEvent(event);
+      document.body.removeChild(a);
+    });  
+  };
+  
+  const download = (image, { name = "DiagnosticDevices", extension = "jpg" } = {}) => {
+    const a = document.createElement("a");
+    a.href = image;
+    //a.download = createFileName(extension, name);
+    a.download = name + "." + extension;
+    document.body.appendChild(a);
+    // wait for the link to be added to the document
+    window.requestAnimationFrame(function () {
+      var event = new MouseEvent('click');
+      a.dispatchEvent(event);
+      document.body.removeChild(a);
+    }); 
+  };
+
+  const downloadScreenshot = () => takeScreenshot(useReactRef.current).then(download);
+
+
+
+
+  useEffect(() => async (dispatch) =>  {
+    /*viewShotRef.current.capture().then((uri) => {
+      console.log('do something with ', uri);
+    });*/
+
+    /*captureRef(viewShotRef, {
+      format: "jpg",
+      quality: 0.8
+    }).then(
+      uri => downLoadImageUrl(uri),
+      error => console.error("Oops, snapshot failed", error)
+    );*/
+
+
+
+  },[])  
+  
+  function downLoadDeviceImage() {
+    console.log("devicesContainer..", devicesContainer);
+    console.log("rootContainer..", rootContainer);
+    
+    html2canvas(rootContainer).then(myCanvas => {
+      document.body.appendChild(myCanvas);
+      deviceImageUrl = myCanvas.toDataURL("image/png", 1.0);
+    }); 
+  
+    const a = document.createElement("a");
+    a.href = deviceImageUrl;
+    a.download = "DiagnosticDevices.png";
+    document.body.appendChild(a);
+    // wait for the link to be added to the document
+    window.requestAnimationFrame(function () {
+      var event = new MouseEvent('click');
+      a.dispatchEvent(event);
+      document.body.removeChild(a);
+    });  
+  
+  
+  };
+  
+
   const handleClose = () => {
     setOpen(!open);
   };
@@ -74,14 +171,17 @@ function DiagnosticsDevices({
   micStatus =  (micValue > 0)?true:false;
   speakerStatus =  (speakerValue > 0)?true:false;
   browserVersion = window.localStorage.getItem('browserVersion'); 
+      /*<ViewShot ref={viewShotRef} options={{ format: 'jpg', quality: 0.9 }}>
+      </ViewShot>*/
 
   return (
-    <div className="container">
+    <div  ref={useReactRef} className="container">
       <Dialog
         open={open}
         onClose={handleClose}
         aria-labelledby="alert-dialog-title"
         aria-describedby="alert-dialog-description"
+        ref={devicesContainer}
       >
         <DialogTitle id="alert-dialog-title">
           <div className="flex-container-top">
@@ -284,7 +384,16 @@ function DiagnosticsDevices({
           />
         </div>
 
-          </div>
+        <div className="flex-child">
+          <input
+            type="submit"
+            size="small"
+            value="Download HTML2CANVAS"
+            onClick={downLoadDeviceImage}
+          />
+        </div>
+
+        </div>
           <br/>
           <br/>
         </Card>
