diff --git a/gaudim/webrtc-client/src/components/HeaderComponent/AppBar.js b/gaudim/webrtc-client/src/components/HeaderComponent/AppBar.js
index db9b35d6c6..3182c04c7b 100644
--- a/gaudim/webrtc-client/src/components/HeaderComponent/AppBar.js
+++ b/gaudim/webrtc-client/src/components/HeaderComponent/AppBar.js
@@ -39,9 +39,11 @@ import { DoRegister } from '@exotel/webrtc-sdk-diagnostics/src/webrtc-client-sdk
 import { UnRegister } from '@exotel/webrtc-sdk-diagnostics/src/webrtc-client-sdk/api/registerAPI/RegisterListener';
 import { DoDiagnostics } from '@exotel/webrtc-sdk-diagnostics/src/webrtc-client-sdk/api/omAPI/DiagnosticsListener';
 import { DiagnosticsSpeakerTestStart } from '@exotel/webrtc-sdk-diagnostics/src/webrtc-client-sdk/api/omAPI/DiagnosticsListener';
+import { DiagnosticsSpeakerTestStop } from '@exotel/webrtc-sdk-diagnostics/src/webrtc-client-sdk/api/omAPI/DiagnosticsListener';
 import { DiagnosticsSpeakerTestStopSuccess } from '@exotel/webrtc-sdk-diagnostics/src/webrtc-client-sdk/api/omAPI/DiagnosticsListener';
 import { DiagnosticsSpeakerTestStopFailure } from '@exotel/webrtc-sdk-diagnostics/src/webrtc-client-sdk/api/omAPI/DiagnosticsListener';
 import { DiagnosticsMicTestStart } from '@exotel/webrtc-sdk-diagnostics/src/webrtc-client-sdk/api/omAPI/DiagnosticsListener';
+import { DiagnosticsMicTestStop } from '@exotel/webrtc-sdk-diagnostics/src/webrtc-client-sdk/api/omAPI/DiagnosticsListener';
 import { DiagnosticsMicTestStopSuccess } from '@exotel/webrtc-sdk-diagnostics/src/webrtc-client-sdk/api/omAPI/DiagnosticsListener';
 import { DiagnosticsMicTestStopFailure } from '@exotel/webrtc-sdk-diagnostics/src/webrtc-client-sdk/api/omAPI/DiagnosticsListener';
 import { ExotelWebClient } from '@exotel/webrtc-sdk-diagnostics/src/webrtc-client-sdk/listeners/ExWebClient';
@@ -52,9 +54,9 @@ import { registerCallback } from '@exotel/webrtc-sdk-diagnostics/src/webrtc-clie
 import { sessionCallback } from '@exotel/webrtc-sdk-diagnostics/src/webrtc-client-sdk/listeners/Callback';
 import { diagnosticsCallback } from '@exotel/webrtc-sdk-diagnostics/src/webrtc-client-sdk/listeners/Callback';
 import logoSrc from '../../static/exotelLogo.jpg'
-import { createRef, useState } from 'react'
-import { useScreenshot, createFileName  } from 'use-react-screenshot'
 import { strictEqual } from 'assert';
+import { createRef, useState } from 'react'
+import html2canvas from "html2canvas";
 
 export default function PrimaryPhoneAppBar(myPhoneDetails) {
   const classes = styles();
@@ -84,15 +86,14 @@ export default function PrimaryPhoneAppBar(myPhoneDetails) {
   const [speakerValue, setSpeakerValue] = React.useState(window.localStorage.getItem('speakerValue'));
   const [micUserResponse, setMicResponse] = React.useState("");
   const [speakerResponse, setSpeakerResponse] = React.useState(window.localStorage.getItem(''));
-
-
+  const [ troubleShootData, setTroubleshootdata ] = React.useState(window.localStorage.getItem('troublrshootData'));
   const imageRef = createRef(null);
   var imageUrl;
   var troubleShootLogs;
-  const [image, takeScreenShot] = useScreenshot({
+  /*const [image, takeScreenShot] = useScreenshot({
     type: "image/jpeg",
     quality: 1.0
-  });
+  });*/
 
   //const numTabs = window.localStorage.getItem('noOfTab') || [];
   console.log("PrimaryPhoneAppBar: User given Phone Details " , myPhoneDetails );
@@ -367,20 +368,14 @@ export default function PrimaryPhoneAppBar(myPhoneDetails) {
     //setReflexFlag
   }
   function micStatusCallback(micStatus, micDescription) {
-    //console.log("Storing key/value: key:" + toString("micValue") + " value:" + toString(micStatus))
-    //console.log("Storing key/value: key:" + toString("micDescr") + " value:" + toString(micDescription))
     window.localStorage.setItem("micValue", micStatus);    
     window.localStorage.setItem("micDescr", micDescription);    
     setMicValue(micStatus);
-    //setMicStatusFlag
   }
   function speakerStatusCallback(speakerStatus, speakerDescription) {
-    //console.log("Storing key/value: key:" + toString("speakerValue") + " value:" + toString(speakerStatus))
-    //console.log("Storing key/value: key:" + toString("speakerDescr") + " value:" + toString(speakerDescription))
     window.localStorage.setItem("speakerValue", speakerStatus);    
     window.localStorage.setItem("speakerDescr", speakerDescription); 
     setSpeakerValue(speakerStatus);    
-    //setSpeakerStatusFlag
   }
   function saveDiagnosticsCallback(saveDiagStatus, saveDiagDescription) {
     //setSaveStatusFlag
@@ -411,7 +406,6 @@ export default function PrimaryPhoneAppBar(myPhoneDetails) {
     /* Start Diagnostics Parallely for the initial screen */ 
     resetDiagnosticValues();
     sendDiagnosticsRequest(); 
-
     setDiagnosticsAsk(false);
     setDiagnosticsDevices(false);    
     setDiagnosticsOps(true);
@@ -450,12 +444,6 @@ export default function PrimaryPhoneAppBar(myPhoneDetails) {
 
 
     const handleDiagnosticsReport= () => {
-      console.log("Setting Diagnostics flag to false");
-
-      //setDiagnosticsReport(false);
-      setDiagnosticsOps(false);  
-      setDiagnosticsAsk(false);           
-      setDiagnosticsDevices(false);
       console.log("Setting Diagnostics flag to true");
       setDiagnosticsReport(true);
     }
@@ -465,9 +453,7 @@ export default function PrimaryPhoneAppBar(myPhoneDetails) {
      * @param {*} event 
      */
     const handlePhoneLogin = (event) => {
-        //setPhoneLoginOpen(!isPhoneLogin);
         setPhoneLoginOpen(true);
-        //handlePhoneLogin(true);
         sendAutoRegistration(event.currentTarget.name)
         setPhoneName(event.currentTarget.name);
         setAnchorPhoneLoginEl(event.currentTarget);
@@ -946,12 +932,16 @@ function makeCall() {
 
 
 const handleDiagnosticsOpsStart = (e) => {
+  DiagnosticsMicTestStop();
+  DiagnosticsSpeakerTestStop();
   resetDiagnosticValues();
   handleDiagnosticsOps();  
   //sendDiagnosticsRequest();  
 }
 
 const handleDiagnosticsDevicesStart  = (e) => {
+  DiagnosticsMicTestStop();
+  DiagnosticsSpeakerTestStop();
   DiagnosticsMicTestStart()
 }
 
@@ -1016,8 +1006,8 @@ function makeTextFile (text) {
 
     return textFile;
 };  
-
-const handleDiagnosticsOpsSaveLogs = (e) => (async() => {
+const rootContainer = document.getElementById("root");
+const handleDiagnosticsOpsSaveLogs = (e)   => {
   e.preventDefault();
   const configData = {
   }
@@ -1038,15 +1028,19 @@ const handleDiagnosticsOpsSaveLogs = (e) => (async() => {
     document.body.removeChild(link);
   });
 
-  console.log("image container..", imageRef.current);
-  canvas = await html2canvas(imageRef);
-  imageUrl = canvas.toDataURL("image/png", 1.0);  
+  /*console.log("image container..", imageRef.current);
+  console.log("rootContainer..", rootContainer);
+  await html2canvas(rootContainer).then(canvas => {
+    document.body.appendChild(canvas);
+    imageUrl = canvas.toDataURL("image/png", 1.0);
+    console.log("imageUrl = ", imageUrl)      
+  });*/
   //window.location.href = window.location.pathname;
   handleDiagnosticsReport();
-})
+}
 
 
-const handleDiagnosticsDevicesSaveLogs = (e)  => (async() =>  {
+const handleDiagnosticsDevicesSaveLogs = (e)  =>  {
   e.preventDefault();
   const configData = {
   }
@@ -1059,7 +1053,6 @@ const handleDiagnosticsDevicesSaveLogs = (e)  => (async() =>  {
   link.setAttribute('download', 'DeviceDiagnostics.txt');
   link.href = makeTextFile(troubleShootLogs);
   document.body.appendChild(link);
-
   // wait for the link to be added to the document
   window.requestAnimationFrame(function () {
     var event = new MouseEvent('click');
@@ -1067,12 +1060,16 @@ const handleDiagnosticsDevicesSaveLogs = (e)  => (async() =>  {
     document.body.removeChild(link);
   });
 
-  console.log("image container..", imageRef.current);
-  canvas = await html2canvas(imageRef);
-  imageUrl = canvas.toDataURL("image/png", 1.0);    
+  /*console.log("image container..", imageRef.current);
+  console.log("rootContainer..", rootContainer);
+  await html2canvas(rootContainer).then(canvas => {
+    document.body.appendChild(canvas);
+    imageUrl = canvas.toDataURL("image/png", 1.0);
+    console.log("imageUrl = ", imageUrl)      
+  });*/   
   //window.location.href = window.location.pathname;
   handleDiagnosticsReport();
-})
+}
 
   function renderPhoneLogin (){
       return  (
